tests/inject/dynamic/shadow-dom.tests.ts

```diff
@@ -88,4 +88,67 @@ describe('Handle Shadow-DOM', () => {
         expect(getComputedStyle(shadowRoot.querySelector('p')).color).toBe('rgb(255, 26, 26)');
     });
 
+    it('should handle defined elements', async () => {
+        container.innerHTML = multiline(
+            '<svg-container>',
+            '   <svg viewBox="0 0 100 30">',
+            '       <style class="testcase-style">#svg-text { color: red; }</style>',
+            '       <text id="svg-text" y="20">I am SVG</text>',
+            '   </svg>',
+            '</svg-container>',
+        );
+        class SVGContainer extends HTMLElement {
+            constructor() {
+                super();
+                const shadowRoot = this.attachShadow({mode: 'open'});
+                const style = document.createElement('style');
+                style.textContent = 'p { color: pink }';
+                const text = document.createElement('p');
+                text.textContent = 'Some text content that should be pink.';
+
+                shadowRoot.append(style);
+                shadowRoot.append(text);
+            }
+        }
+        customElements.define('svg-container', SVGContainer);
+
+        createOrUpdateDynamicTheme(theme, null, false);
+
+        await timeout(100);
+        const shadowRoot = document.querySelector('svg-container').shadowRoot;
+        expect(getComputedStyle(shadowRoot.querySelector('p')).color).toBe('rgb(255, 198, 208)');
+    });
+
+    it('should react to defined elements', async () => {
+        container.innerHTML = multiline(
+            '<svg-container-2>',
+            '   <svg viewBox="0 0 100 30">',
+            '       <style class="testcase-style">#svg-text { color: red; }</style>',
+            '       <text id="svg-text" y="20">I am SVG</text>',
+            '   </svg>',
+            '</svg-container2>',
```

Comment:
Pay attention that tags don't match. Use the same suggestions as for a test above.